### DISCUSSION TITLE
add http-server.authentication.type=PASSWORD cause error

### DISCUSSION THREAD

**Discussion by s-amir** [trinodb/trino#23227] [Q&A]
I try to connect trino to LDAP , so in config.properties i add "http-server.authentication.type=PASSWORD" to authenticate users with LDAP
i also create **password-authenticator.properties** in **/etc/password-authenticator.properties**
but as I append "**http-server.authentication.type=PASSWORD**" in config.properties i get error : 
`Invalid configuration property with prefix '': file does not exist: etc/password-authenticator.properties (for class io.trino.server.security.PasswordAuthenticatorConfig.passwordAuthenticatorFiles[0].<list element>)`

can anyone help

**Comment by hashhar** [comment]
What did you use to deploy? Docker image or tarball?

It seems you have placed the config file at `/etc/password-authenticator.properties`. If you re-read the docs you'll notice it says `etc/password-authenticator.properties` (note the lack of leading `/`) which means the path is relative to the Trino data directory. For the Docker image this means `/etc/trino` while in the tarball this means the same directory where you extracted Trino itself.

**Reply by s-amir** [reply]
thanks, i changed the path to what you said and i get different error 
_IllegalArgumentException: Shared secret (internal-communication.shared-secret) is required when authentication is enabled_
can you help me how to solve it ?

password-authenticator.properties:
```
password-authenticator.name=ldap
ldap.url=ldaps://*******:636
ldap.ssl.truststore.path=/etc/trino/certs/trino-server.jks
ldap.ssl.truststore.password=changeit
ldap.user-bind-pattern=CN=%s,OU=data,DC=blu,DC=ir
```

I don't no what is shared-key!

**Reply by hashhar** [reply]
Add `internal-communication.shared-secret=<some-random-but-unguessable-value>` to `config.properties` on all nodes. 
See https://trino.io/docs/current/security/overview.html#:~:text=Configure%20a%20a%20shared%20secret for the details.

See https://trino.io/docs/current/security/overview.html#suggested-configuration-workflow for the overall steps which I think you haven't seen yet.

**Reply by s-amir** [reply]
Thank you very much , I have another qustion , I want to use self-signed key , in this approach should I my self generate keys and certificate ?
finally I want to enable TLS in trino to connect to LDAP , and should I copy any certificate from ldap to trino ?
I read the documentation but i don't find any document to generate self-signed key!
I really appreciate your answers

**Reply by hashhar** [reply]
See https://trino.io/docs/current/security/tls.html#secure-trino-directly for TLS in general. Self-signed certs are no different. You import then into a trust/keystore and follow the same steps.

However using self-signed instead of public certs or at-least using an internal CA is asking for a lot of pain.

**Reply by yogesh-kalkate** [reply]
@hashhar i update changes as you mentioned in previous reply now my application ask for authentication but I deployed it on cluster with 1 coordinator and 2 worker config , now issue is that worker pod was not coming up they showing following error
I used its official helm chart for deploy 

Errors:

1) Invalid configuration property with prefix '': file does not exist: etc/password-authenticator.properties (for class io.trino.server.security.PasswordAuthenticatorConfig.passwordAuthenticatorFiles[0].<list element>)

I update in values.yaml like this I'm using password authentication

  authenticationType: "PASSWORD"
    query:
      maxMemory: "8GB"
  exchangeManager:
    name: "filesystem"
    baseDir: "/tmp/trino-local-file-system-exchange-manager"
  workerExtraConfig: "internal-communication.shared-secret=test"
  coordinatorExtraConfig: "internal-communication.shared-secret=test"
  
  
  additionalConfigProperties:
- http-server.process-forwarded=true
- http-server.authentication.type=PASSWORD
- internal-communication.https.required=true
- internal-communication.shared-secret=test
- http-server.https.enabled=true

I tried with volume mount, secret as well

**Reply by hashhar** [reply]
As the error message says you don't have a `password-authenticator.properties` at all (or at the correct location). From the partial values you have shared it seems you don't define a config-map/mount for that file.

Please ask on #kubernetes on the Trino slack.

**Reply by yogesh-kalkate** [reply]
@hashhar But I mentioned this in a global section which is applicable for both workers and coordinator , currently my coordinator pod was running fine, this properties file was correctly located inside the coordinator container .

**Reply by hashhar** [reply]
then make the file available on workers too as the error suggests.
